src/components/task2/drawer.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from "react";
+import "./style.scss";
+
+
+export default class Drawer extends React.Component {
+    constructor(props) {
+      super(props);
+  
+      this.state = { expanded: false };
+    }
+  
+    onToggleButtonClick() {
+      this.setState(state => ({ expanded: !state.expanded }));
+    }
+  
+    render() {
+      return (
+        <div>
+          <button onClick={this.onToggleButtonClick.bind(this)}></button>
+          { !this.state.expanded ?
+               (this.renderDrawer()) : (null)
+          }
+        </div>
```

Comment:
This is not the right solution to this problem. You use the ternary operator if you want to reply 2 different results on `true` or `false`. If you want to return `null` on `false` then the best solution is to use the `&&` operator.

```javascript
{ this.state.expanded && this.renderDrawer() }
```